test/CMakeLists.txt

```diff
@@ -2,6 +2,55 @@
 
 include(BoostTestTargets)
 
+set(types_test_SOURCES
+  icingaapplication-fixture.cpp
+  base-type.cpp
+  ${base_OBJS}
+  $<TARGET_OBJECTS:config>
+  $<TARGET_OBJECTS:remote>
+  $<TARGET_OBJECTS:icinga>
+  $<TARGET_OBJECTS:methods>
+)
+
+if(ICINGA2_WITH_CHECKER)
+  list(APPEND types_test_SOURCES $<TARGET_OBJECTS:checker>)
+endif()
+
+if(ICINGA2_WITH_MYSQL)
+  list(APPEND types_test_SOURCES $<TARGET_OBJECTS:db_ido> $<TARGET_OBJECTS:db_ido_mysql>)
+endif()
+
+if(ICINGA2_WITH_PGSQL)
+  list(APPEND types_test_SOURCES $<TARGET_OBJECTS:db_ido> $<TARGET_OBJECTS:db_ido_pgsql>)
+endif()
+
+if(ICINGA2_WITH_ICINGADB)
+  list(APPEND types_test_SOURCES $<TARGET_OBJECTS:icingadb>)
+endif()
+
+if(ICINGA2_WITH_NOTIFICATION)
+  list(APPEND types_test_SOURCES $<TARGET_OBJECTS:notification>)
+endif()
+
+if(ICINGA2_WITH_PERFDATA)
+  list(APPEND types_test_SOURCES $<TARGET_OBJECTS:perfdata>)
+endif()
+
+if(ICINGA2_UNITY_BUILD)
+  mkunity_target(types test types_test_SOURCES)
+endif()
+
+add_boost_test(types
+  SOURCES test-runner.cpp ${types_test_SOURCES}
+  LIBRARIES ${base_DEPS}
+  TESTS
+    types/gettype
+    types/assign
+    types/byname
+    types/instantiate
+    types/sort_by_load_after
+)
```

Comment:
There should at least be a comment what this does and why it's necessary. I'd say this is very non-obvious.